cni/pkg/install/install.go

```diff
@@ -189,16 +211,20 @@ func (in *Installer) sleepCheckInstall(ctx context.Context, installedBinFiles se
 	}()
 
 	for {
-		if err := checkInstall(in.cfg, in.cniConfigFilepath); err != nil {
-			// Pod set to "NotReady" due to invalid configuration
-			installLog.Infof("Invalid configuration. %v", err)
+		// Before we process whether any file events have been triggered, we must check that the file is correct
+		// at this moment, and if not, yield. This is to catch other CNIs which might have mutated the file between
+		// the (theoretical) window after we initially install/write, but before we actually start the filewatch.
+		if err := checkValidCNIConfig(in.cfg, in.cniConfigFilepath); err != nil {
```

Comment:
what if the kubeconfig or binaries changed? Can't those have happened before we start the watcher

Reply:
1. They can. But it's exceedingly unlikely - unlike the CNI config, nothing else but us cares about those files.
2. If someone injected bad binaries and/or kubeconfig, they'd have a remarkably tiny window to do it in, and they'd have to have broad privilege to the host node's root filesystem already - and already effectively have the ability to hijack the entire node CNI system, so effectively the node would already be irrevocably compromised.

Reply:
I was thinking more if another CNi is too aggressive and wipes our files. But then I guess they would probably do so again once we wrote them... so agree its probably not a concern.

Plus its been like this forever without issues